java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -0,0 +1,300 @@
+      if (isNull != right.isNull(rightIndex)) {
+        return false;
+      }
+
+      int offsetWidth = BaseVariableWidthVector.OFFSET_WIDTH;
+
+      if (!isNull) {
+        final int start1 = left.getOffsetBuffer().getInt(leftIndex * offsetWidth);
+        final int end1 = left.getOffsetBuffer().getInt((leftIndex + 1) * offsetWidth);
+
+        final int start2 = right.getOffsetBuffer().getInt(rightIndex * offsetWidth);
+        final int end2 = right.getOffsetBuffer().getInt((rightIndex + 1) * offsetWidth);
+
+        int ret = ByteFunctionHelpers.equal(left.getDataBuffer(), start1, end1,
+            right.getDataBuffer(), start2, end2);
+
+        if (ret == 0) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  protected boolean compareListVectors(ListVector left) {
+    if (!left.getField().getType().equals(right.getField().getType())) {
+      return false;
+    }
+
+    for (int i = 0; i < length; i++) {
+      int leftIndex = leftStart + i;
+      int rightIndex = rightStart + i;
+
+      boolean isNull = left.isNull(leftIndex);
+      if (isNull != right.isNull(rightIndex)) {
+        return false;
+      }
+
+      int offsetWidth = BaseRepeatedValueVector.OFFSET_WIDTH;
+
+      if (!isNull) {
+        final int start1 = left.getOffsetBuffer().getInt(leftIndex * offsetWidth);
+        final int end1 = left.getOffsetBuffer().getInt((leftIndex + 1) * offsetWidth);
+
+        final int start2 = right.getOffsetBuffer().getInt(rightIndex * offsetWidth);
+        final int end2 = right.getOffsetBuffer().getInt((rightIndex + 1) * offsetWidth);
+
+        if ((end1 - start1) != (end2 - start2)) {
+          return false;
+        }
+
+        ValueVector dataVector1 = left.getDataVector();
+        ValueVector dataVector2 = ((ListVector)right).getDataVector();
+
+        if (!dataVector1.accept(new RangeEqualsVisitor(dataVector2, start1, start2, (end1 - start1)))) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  protected boolean compareFixedSizeListVectors(FixedSizeListVector left) {
+    if (!left.getField().getType().equals(right.getField().getType())) {
+      return false;
+    }
+
+    if (left.getListSize() != ((FixedSizeListVector)right).getListSize()) {
+      return false;
+    }
+
+    for (int i = 0; i < length; i++) {
+      int leftIndex = leftStart + i;
+      int rightIndex = rightStart + i;
+
+      boolean isNull = left.isNull(leftIndex);
+      if (isNull != right.isNull(rightIndex)) {
+        return false;
+      }
+
+      int listSize = left.getListSize();
+
+      if (!isNull) {
+        final int start1 = leftIndex * listSize;
+        final int end1 = (leftIndex + 1) * listSize;
+
+        final int start2 = rightIndex * listSize;
+        final int end2 = (rightIndex + 1) * listSize;
+
+        if ((end1 - start1) != (end2 - start2)) {
+          return false;
+        }
+
+        ValueVector dataVector1 = left.getDataVector();
```

Comment:
please don't use 1 and 2 in variable names..

Reply:
Thanks, fixed.